### DISCUSSION TITLE
AVIF Missing on Linux mogrify

### DISCUSSION THREAD

**Discussion by Sondro** [ImageMagick/ImageMagick#5241] [Help]
### ImageMagick version

7.1.0-37

### Operating system

Linux

### Operating system, version and so on

Debian Bullseye

### Description

Tried portable version on the main site + gcc & clang AppImage versions (using the direct paths).

I'm able to use mogrify to convert to various formats like png, webp, jpg, but avif returns an 'unrecognized format error' like:

**/pathTo/ImageMagick--gcc-x86_64.AppImage mogrify -format avif \*.png
mogrify: unable to open image 'avif:': No such file or directory @ error/blob.c/OpenBlob/3533.
mogrify: unrecognized image format 'avif' @ error/mogrify.c/MogrifyImageCommand/5132.**

AVIF is also missing from the list when trying: `identify -list format`.

I'm also able to return the correct version for each like:

**pathTo/ImageMagick--gcc-x86_64.AppImage -version
Version: ImageMagick 7.1.0-37 Q16-HDRI x86_64 b65d0d9af:20220605 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): bzlib djvu fontconfig freetype jbig jng jpeg lcms lqr lzma openexr png raqm tiff webp x xml zlib
Compiler: gcc (7.5)**

### Steps to Reproduce

List missing AVIF:

`/pathTo/magick identify -list format `

`/pathTo/ImageMagick--clang-x86_64.AppImage identify -list format`

`/pathTo/ImageMagick--gcc-x86_64.AppImage identify -list format`

==============================================================

Unrecognized format:

`/pathTo/magick mogrify -format avif *.png`

`/pathTo/ImageMagick--clang-x86_64.AppImage mogrify -format avif *.png`

`/pathTo/ImageMagick--gcc-x86_64.AppImage mogrify -format avif *.png`

**Comment by dlemstra** [comment]
This is happening because libheif is not build with AVIF support.

> For AVIF, libaom, dav1d, or rav1e are used as codecs.

That quote is from here: https://github.com/strukturag/libheif.

**Comment by Sondro** [comment]
I can use the UI to save AVIFs so why is it supported in the UI, but not the command line?

Do I have to recompile Image Magick after installing AVIF codecs for libheif?

**Comment by dlemstra** [comment]
Which UI are you talking about?

**Comment by Alvie** [comment]
Not just `mogrify` but `convert` also. The AppImage does not delegate `heic`.

It seems when `magick test.png test.avif` is run, it creates an AVIF file but it is actually just a PNG with the .avif extension:
```
~$ identify AppImage.avif Compiled.avif
AppImage.avif PNG 7084x2105 7084x2105+0+0 8-bit sRGB 25.1257MiB 0.000u 0:00.000
Compiled.avif AVIF 7084x2105 7084x2105+0+0 8-bit sRGB 1.37931MiB 0.010u 0:00.000
```
I apologise if this is a separate issue, I can create a new one if you believe so? Or if it is not an issue but intended behaviour, can you clarify whether or not the AppImage is supposed to produce .avif files with the AVIF type?

**Comment by dlemstra** [comment]
I did not realize that the appimage was used. There is no support for `avif` in the appimage build but we are open to someone creating a pull request to add support for this.

**Reply by Alvie** [reply]
Thanks for letting us know, that makes a lot of sense now. I have manually compiled it myself now and it works fine. It would be nice to have it in .AppImage as it seems like most distributions only have version 6.9 but AVIF is only supported in version 7.0.25 or later.

A PPA with the latest versions for debian/Ubuntu based systems that supports AVIF would be even better (if it doesn't exist already, if so I apologize but the information is hard to find).

Thanks though, I will run it compiled manually for now as it does work great so personally I'm not too worried about the AppImage not supporting AVIF but it was just a bit confusing.
